relay-metrics/src/bucket.rs

```diff
@@ -748,33 +749,55 @@ pub struct BucketMetadata {
     /// For example: Merging two un-merged buckets will yield a total
     /// of `2` merges.
     pub merges: NonZeroU32,
+
+    /// Received timestamp of the first metric in this bucket.
+    ///
+    /// This field should be set to the time in which the first metric of a specific bucket was
+    /// received in the outermost internal Relay.
+    pub received_at: Option<UnixTimestamp>,
 }
 
 impl BucketMetadata {
     /// Creates a fresh metadata instance.
     ///
-    /// The new metadata is initialized with `1` merge.
-    pub fn new() -> Self {
+    /// The new metadata is initialized with `1` merge and a given `received_at` timestamp.
+    pub fn new(received_at: UnixTimestamp) -> Self {
         Self {
             merges: NonZeroU32::MIN,
+            received_at: Some(received_at),
         }
     }
 
     /// Whether the metadata does not contain more information than the default.
     pub fn is_default(&self) -> bool {
-        let Self { merges } = self;
-        *merges == NonZeroU32::MIN
+        let Self {
+            merges,
+            received_at,
+        } = self;
+
+        *merges == NonZeroU32::MIN && received_at.is_none()
     }
 
     /// Merges another metadata object into the current one.
     pub fn merge(&mut self, other: Self) {
         self.merges = self.merges.saturating_add(other.merges.get());
+        self.received_at = match (self.received_at, other.received_at) {
+            (Some(received_at), None) => Some(received_at),
+            (None, Some(received_at)) => Some(received_at),
+            (Some(left_received_at), Some(right_received_at)) => {
+                Some(min(left_received_at, right_received_at))
+            }
+            (None, None) => None,
+        };
     }
 }
 
 impl Default for BucketMetadata {
     fn default() -> Self {
-        Self::new()
+        Self {
+            merges: NonZeroU32::MIN,
+            received_at: None,
+        }
     }
 }
```

Comment:
I don't think we need default here, let's move it to a separate constructor, maybe empty if we need it somewhere. But generally there shouldn't be any instances where we create a bucket without timestamp.